src/ffi.rs

```diff
@@ -167,6 +168,193 @@ pub const X509_V_ERR_UNSUPPORTED_EXTENSION_FEATURE: c_int = 45;
 pub const X509_V_ERR_UNSUPPORTED_NAME_SYNTAX: c_int = 53;
 pub const X509_V_OK: c_int = 0;
 
+pub mod nid {
+    use std::mem;
+    use std::collections::enum_set::CLike;
+
+    #[allow(dead_code)]
+    #[allow(non_camel_case_types)]
+    #[repr(uint)]
+    pub enum Nid {
+        Undefined,
```

Comment:
I'm not convinced that's the right location for NID values. I ran into an issue trying to get text for `Nid::UID` from the subject, and changing the UID value to the one defined in `include/openssl/obj_mac.h` resolved the problem.